rust/rust-sysdb/src/server.rs

```diff
@@ -153,23 +156,80 @@ impl SysDb for SysdbService {
 
     async fn create_tenant(
         &self,
-        _request: Request<CreateTenantRequest>,
+        request: Request<CreateTenantRequest>,
     ) -> Result<Response<CreateTenantResponse>, Status> {
-        todo!()
+        let proto_req = request.into_inner();
+        // Convert proto -> internal request type
+        let internal_req: internal::CreateTenantRequest = proto_req.into();
+
+        let backends = self.router.route(&RouteRequest {
+            op: Operation::CreateTenant,
+            db_name: None,
+            tenant: Some(&internal_req.name),
+        })?;
+
+        // Fan out to all backends
+        let mut internal_resp = None;
+        for backend in backends {
+            internal_resp = Some(backend.create_tenant(&internal_req).await?);
+        }
+
+        // Convert internal -> proto response type
+        let resp = internal_resp.ok_or_else(|| Status::internal("no backend for request"))?;
+        Ok(Response::new(resp.into()))
     }
 
     async fn get_tenant(
         &self,
-        _request: Request<GetTenantRequest>,
+        request: Request<GetTenantRequest>,
     ) -> Result<Response<GetTenantResponse>, Status> {
-        todo!()
+        let proto_req = request.into_inner();
+        // Convert proto -> internal request type
+        let internal_req: internal::GetTenantRequest = proto_req.into();
+
+        let backends = self.router.route(&RouteRequest {
+            op: Operation::GetTenant,
+            db_name: None,
+            tenant: Some(&internal_req.name),
+        })?;
+
+        // Should be exactly one backend for GetTenant
```

Comment:
This abstraction feels leaky: I shouldn't know about what backends I am talking to here